### ISSUE TITLE
How to restore orderbook after "invalid checksum"? (bitget)

### ISSUE THREAD

**Issue by interenet** [ccxt/ccxt#23453]
### Operating System

Ubuntu 22.04

### Programming Languages

Python

### CCXT Version

Ccxt 4.2.25

### Description

(question)
After catch fault it will repeat. How to rerun and get new snapshot of orderbook? Thanks

### Code

```
async def watch_orderbook(exchange):
    while True:
        try:
            orderbook = await exchange.watch_orderbook('BTC/USDT')
         except Exception:
             continue
```

**Comment by carlosmiei** [comment]
Hello @interenet, catching and continuing as you're doing should be enough, we will take a look. In the meantime, can you try to do `await exchange.close()` inside the `except` branch?

**Comment by interenet** [comment]
No, i can not perform echange.close(). - i have several tasks in gather  (balance, trades).
If i catche fault "invalid checksum" in exception it stayes forever.

**Comment by carlosmiei** [comment]
@interenet I see, while we work on a fix, can you temporarily disable the checksum? `exchange.options['checksum'] = True`

**Comment by interenet** [comment]
Yes, i can use this option. But i will get incorrect prices and amounts.
A will wait fix.
Thanks

**Comment by carlosmiei** [comment]
Hello @interenet, we just released a version with a fix, now if you get a checksum error the next watchOrderBook call will restart the subscription from scratch.

Can you please upgrade and try again?

**Comment by interenet** [comment]
It looks great. Thanks

**Comment by interenet** [comment]
After tests i have new strange fault "= False" in watch_orderbook. What does it mean? Its all text of fault.

**Comment by carlosmiei** [comment]
@interenet Can you share some logs please? how to reproduce the issue?

**Comment by interenet** [comment]
You can run 10 this task for different symbols and wait about two days. In logs i had only "= False" about 10 times

`async def watch_orderbook(exchange):
    while True:
        try:
            orderbook = await exchange.watch_orderbook('BTC/USDT')
         except Exception as e:
             print (e)
             continue`

**Comment by carlosmiei** [comment]
@interenet Is there a quicker way? waiting 2 days is tough 😅

**Comment by kroitor** [comment]
Hello @interenet 👋

Doing a cleanup of older issues — the original question here was resolved back in Aug 2024 (watch_order_book now automatically restarts the subscription from scratch after an "invalid checksum" error, which you confirmed was working 🎉).

The remaining "= False" fault was never something we could reproduce without logs, and a lot has changed in the WS layer since 4.2.x, so I'll close this one as inactive. If you ever run into that error again on a recent version, please open a fresh issue with the verbose output (`exchange.verbose = True`) and we'll dig in right away.

Thanks for the patience and the detailed back-and-forth! 🙏
